views/default/elgg/admin.js

```diff
@@ -60,12 +61,55 @@ define(function(require) {
 		
 		// plugin details selection
 		$(document).on('click', '.elgg-plugin-details-container > ul a', showPluginDetails);
-		
+
+		$(document).on('click', '.elgg-plugins-toggle', toggleAllPlugins);
+
 		// plugin screenshots
 		$(document).on('mouseenter', '.elgg-plugin-details-screenshots .elgg-plugin-screenshot', showPluginScreenshot);
 	}
 
 	/**
+	 * Active or deactivate all the visible plugins
+	 *
+	 * @param {Event} e click event
+	 * @return void
+	 */
+	function toggleAllPlugins(e) {
+		e.preventDefault();
+
+		if (!confirm(elgg.echo('question:areyousure'))) {
+			return;
+		}
+
+		var guids = [],
+			state = $(this).data('desiredState'),
+			find_state = state == 'active' ? 'inactive' : 'active';
+
+		$('.elgg-plugin.elgg-state-' + find_state + ':visible').each(function () {
+			var guid = $(this).data('guid');
+			if (guid) {
+				guids.push(guid);
+			}
+		});
+
+		if (!guids.length) {
+			elgg.register_error(elgg.echo('admin:plugins:already:' + state));
+			return;
+		}
+
+		spinner.start();
+
+		// We create a regular form and submit it. This is the simplest way to send the
+		// data, have the page refreshed, and make sure error messages are still shown on
+		// the new page. Using ajax leads to complexity because Elgg wants to send the error
+		// messages back to the client.
```

Comment:
The ajax path worked identically, but required ugly code in the actions to avoid forward() and unregister elgg_ajax_forward().

We could wrap the whole page in a form, and just alter hidden inputs when we alter visibility...and change action URL depending on button...

Just at turning the Ajax into a vanilla form submit seemed least bad.